builds/jenkins-build-pack-api/Jenkinsfile

```diff
@@ -223,45 +227,63 @@ node() {
+							case 'aws_apigateway':
+								events.sendStartedEvent('DEPLOY_TO_AWS_APIGATEWAY', 'Deployment started for API Gateway', context_map, environment_logical_id)
+								def api_id = utilModule.getAPIId(configLoader.AWS.API[env_key], config)
+								echo "API ID: $api_id"
+								apiHostName ="${api_id}.execute-api.${configLoader.AWS.REGION}.amazonaws.com"
+								awsAPIGatewayModule.addApigatewayLambdaIntegration('swagger/swagger.json');
+								updateSwagger(apiHostName, environment_logical_id, current_environment, env_key, config)
+
+								sh "aws apigateway put-rest-api --rest-api-id ${api_id} --mode merge --parameters basepath=prepend --body 'file://swagger/swagger.json'"+ " --profile cloud-api"
+								sh "aws apigateway create-deployment --rest-api-id ${api_id} --stage-name ${current_environment} --profile cloud-api"
+								sh "aws apigateway tag-resource --resource-arn arn:aws:apigateway:${configLoader.AWS.REGION}::/restapis/${api_id}/stages/${current_environment} --tags Application=Jazz,JazzInstance=${configLoader.INSTANCE_PREFIX}  --profile cloud-api"
+								if("${configLoader.JAZZ.API_DETAILED_MONITORING}" == "true") {
+									sh "aws apigateway update-stage --rest-api-id ${api_id} --stage-name ${current_environment} --patch-operations op=replace,path=/*/*/metrics/enabled,value=true --region ${configLoader.AWS.REGION}"
+								} else {
+									sh "aws apigateway update-stage --rest-api-id ${api_id} --stage-name ${current_environment} --patch-operations op=replace,path=/*/*/metrics/enabled,value=false --region ${configLoader.AWS.REGION}"
+								}
+        
+								if (current_environment == 'prod' || config['domain'] == 'jazz') {
+									endpointUrl = "https://${apiHostName}/${current_environment}/${config['domain']}/${config['service']}"
+								} else {
+									endpointUrl = "https://${apiHostName}/${current_environment}/${environment_logical_id}/${config['domain']}/${config['service']}"
+								}
+
+								def apiArns = getAPIGatewayResourceARNs(api_id, environment_logical_id)
+								if(apiArns) {
+									for (arn in apiArns) {
+										events.sendCompletedEvent('CREATE_ASSET', null, utilModule.generateAssetMap("aws", arn, "apigateway", config), environment_logical_id);
+									}
+								}
+								events.sendCompletedEvent('DEPLOY_TO_AWS_APIGATEWAY', 'Deployment completed for API Gateway', context_map, environment_logical_id)
+								break;
+
+							case 'gcp_apigee':
+								events.sendStartedEvent('DEPLOY_TO_GCP_APIGEE', 'Deployment started for APIGEE', context_map, environment_logical_id)
+								def arn = awsLambdaModule.getLambdaARN(stackName)
+								apiHostName ="${configLoader.APIGEE.API_ENDPOINTS[env_key].service_hostname}"
+								updateSwagger(apiHostName, environment_logical_id, current_environment, env_key, config)
+								endpointUrl = apigeeModule.deploy("${pwd()}/swagger/swagger.json", arn, env_key, environment_logical_id, config, context_map)
```

Comment:
This might not be required - ${pwd()}/swagger/swagger.json. We use swagger/swagger.json across the board, we can continue using the same here too.

Reply:
I use full path so we don't have to worry about a mismatch if a downstream operation runs in a dir() block. I can have the function in apigee-module determine the full path on its side, though.